From 1b5c0bda3970c722b3f257266bca9b72dcf6d2b4 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sun, 11 Apr 2021 06:17:38 -0700
Subject: [PATCH] Replace stat call with __xstat

glibc 2.33 reworked the stat() family of syscalls, but we
need this plugin to work inside L4T containers that use
an older version of glibc, so replace stat() with the
older __xstat(0, ...) for compatibility.

Signed-off-by: Matt Madison <matt@madison.systems>
---
 v4l2_calls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/v4l2_calls.c b/v4l2_calls.c
index dea63b4..644050a 100644
--- a/v4l2_calls.c
+++ b/v4l2_calls.c
@@ -550,7 +550,7 @@ gst_v4l2_open (GstV4l2Object * v4l2object)
     }
   } else if (is_cuvid == FALSE) {
     /* check if it is a device */
-    if (stat (v4l2object->videodev, &st) == -1)
+    if (__xstat (0, v4l2object->videodev, &st) == -1)
       goto stat_failed;
 
     if (!S_ISCHR (st.st_mode))
